src/checkoutservice/kafka/trace_interceptor.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package kafka
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/IBM/sarama"
+)
+
+type OTelInterceptor struct {
+	tracer     trace.Tracer
+	fixedAttrs []attribute.KeyValue
+}
+
+// NewOTelInterceptor processes span for intercepted messages and add some
+// headers with the span data.
+func NewOTelInterceptor() *OTelInterceptor {
+	oi := OTelInterceptor{}
+	oi.tracer = otel.Tracer("github.com/open-telemetry/opentelemetry-demo/checkoutservice/sarama")
```

Comment:
I thinks "github.com/IBM/sarama" is better

Reply:
This should be `checkoutservice` to be consistent with how we are naming the Tracer with other parts of this service: https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/checkoutservice/main.go#L159